### DISCUSSION TITLE
How to query all entities with certain components, regardless of group

### DISCUSSION THREAD

**Discussion by Danon5** [sebas77/Svelto.ECS#97] [Q&A]
How would I query all entities with a tag (or without a tag) that have a certain set of components, regardless of their actual group? I get errors if all the entities in the group do not have the same components. It feels as if all entity querying is supposed to query on a specific group, but isn't that super restrictive? I would think that I should be able to iterate over all components of a certain type in the whole simulation. I don't really know my options though, or if I am approaching it the right way.

**Comment by sebas77** [comment]
In this article: https://www.sebaslab.com/ecs-abstraction-layers-and-modules-encapsulation/

check the paragraph

_Some more words about Abstraction Layers with Svelto.ECS_

Svelto provides different ways to query entities according to the layer of abstraction.

Reading this article should clarify when to use which.

FindGroups specifically will solve your problem.

In rare situations, you can still use an EGIDMapper and the entity EGID to look up for other components albeit the operation is slower
